syntax = "proto3";

package user;

// 用户信息
message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
  
  enum UserType {
    NORMAL = 0;
    ADMIN = 1;
    GUEST = 2;
  }
  
  UserType user_type = 4;
  repeated string phone_numbers = 5;
  
  message Address {
    string street = 1;
    string city = 2;
    string country = 3;
  }
  
  Address address = 6;
}

message GetUserRequest {
  int32 user_id = 1;
}

message CreateUserRequest {
  User user = 1;
}

message UserResponse {
  bool success = 1;
  string message = 2;
  User user = 3;
}

// 用户服务
service UserService {
  rpc GetUser(GetUserRequest) returns (UserResponse);
  rpc CreateUser(CreateUserRequest) returns (UserResponse);
} 